pkg/flow/tracer_map.go

```diff
@@ -44,17 +44,18 @@ func NewMapTracer(fetcher mapFetcher, evictionTimeout, staleEntriesEvictTimeout
 		lastEvictionNs:             uint64(monotime.Now()),
 		evictionCond:               sync.NewCond(&sync.Mutex{}),
 		staleEntriesEvictTimeout:   staleEntriesEvictTimeout,
-		hmapEvictionCounter:        m.CreateHashMapCounter(),
-		numberOfEvictedFlows:       m.CreateNumberOfEvictedFlows(),
+		evictionCounter:            m.GetEvictionCounter(),
+		evictedFlowsCounter:        m.GetEvictedFlowsCounter(),
 		timeSpentinLookupAndDelete: m.CreateTimeSpendInLookupAndDelete(),
 		errors:                     m.GetErrorsCounter(),
 	}
 }
 
 // Flush forces reading (and removing) all the flows from the source eBPF map
 // and sending the entries to the next stage in the pipeline
-func (m *MapTracer) Flush() {
+func (m *MapTracer) Flush(reason string) {
 	m.evictionCond.Broadcast()
+	m.evictionCounter.ForSourceAndReason("hashmap", reason).Inc()
```

Comment:
@msherif1234 I think what I did here doesn't work: the eviction counter is incremented everytime `Flush` is called, but it seems to be called many times before sync.Condition makes it go to the actual evict. I'm not sure why and I wonder if this is normal. But to "fix" this counter I would need to move the increment back where you set it before, in `evictFlows`, but the downside is that we loose then the `reason` label that I wanted to have

Reply:
Look at this: there's a peak to 300K evictions / second due to this increase, while the flows per second is only at 3.5K. Isn't it fishy?

![Capture d’écran du 2024-02-27 16-11-25](https://github.com/netobserv/netobserv-ebpf-agent/assets/2153442/be8b13ab-1dc6-4477-bed0-a9251d6c6870)